DancingGoat/Scripts/PreviewMode/PreviewMode.js

```diff
@@ -0,0 +1,107 @@
+﻿const previewModeToggleClassname = 'preview-mode-switch';
+const previewModeCookieName = 'IsPreviewMode';
+const previewApiKeyCookieName = 'PreviewApiKey';
+const enterApiKeyFirstMessage = 'Enter your Preview API key';
+const enterApiKeyNextMessage = 'Enter your Delivery Preview API key -- you can find it in Kentico Cloud under Project settings -> API keys';
+const enterApiKeyPromptTitle = 'Preview API key';
+
+const previewModeToggles = document.getElementsByClassName(previewModeToggleClassname);
+const isPreviewModeEnabled = getCookie(previewModeCookieName);
+for (toggle of previewModeToggles) {
+    toggle.checked = isPreviewModeEnabled === 'true' ? true : false;
+}
+
+function togglePreviewMode(isChecked, projectId, isFirstEnterOfApiKey) {
+    if (isChecked) {
+        const apiKey = getCookie(previewApiKeyCookieName);
+        if (apiKey === '') {
+            enterApiKey(isFirstEnterOfApiKey, projectId);
+        } else {
+            setPreviewModeEnabledCookie(isChecked);
+            location.reload();
+        }
+    } else {
+        setPreviewModeEnabledCookie(isChecked);
+        location.reload();
+    }
+}
+
+function enterApiKey(isFirstEnterOfApiKey, projectId) {
+    logAttemptToEnterPreviewKey(isFirstEnterOfApiKey);
+    const message = isFirstEnterOfApiKey ? enterApiKeyFirstMessage : enterApiKeyNextMessage;
+    const apiKey = prompt(message, enterApiKeyPromptTitle);
+    if (apiKey === null) {
+        const previewModeToggles = document.getElementsByClassName(previewModeToggleClassname);
+        for (toggle of previewModeToggles) {
+            toggle.checked = false
+        }
+    }
+
+    if (apiKey !== null) {
```

Comment:
At this point we can be pretty certain that apiKey is not null, see line 33

Reply:
Do you try to suggest it should be in else block?